tt_metal/hw/ckernels/grayskull/llk_lib/llk_unpack_AB_matmul.h

```diff
@@ -70,41 +62,29 @@ ALWI void llk_unpack_AB_matmul_mop_config(bool transpose) {
     tmp.program(instrn_buffer);
 }
 
-inline void llk_unpack_AB_matmul_hw_configure(const llk_unpack_AB_matmul_params_t *unpack_AB_params) {
+inline void _llk_unpack_AB_matmul_hw_configure_(const std::uint32_t unpA_src_format, const std::uint32_t unpB_src_format, const std::uint32_t unpA_dst_format, const std::uint32_t unpB_dst_format) {
     configure_unpack_AB(
-        get_operand_id(unpack_AB_params->unpB_operand), get_operand_id(unpack_AB_params->unpA_operand), 16, 16);
+        unpA_src_format,
+        unpB_src_format,
+        unpA_dst_format,
+        unpB_dst_format,
+        16, 16
+    );
 }
 
-inline void llk_unpack_AB_matmul_hw_configure_disaggregated(
-    const std::uint32_t unpA_operand, const std::uint32_t unpB_operand, const std::uint32_t transpose_xy_srca = 0) {
-    const llk_unpack_AB_matmul_params_t unpack_AB_matmul_params = {
-        .unpA_operand = unpA_operand, .unpB_operand = unpB_operand, .transpose_xy_srca = transpose_xy_srca};
-    llk_unpack_AB_matmul_hw_configure(&unpack_AB_matmul_params);
+inline void _llk_unpack_AB_matmul_init_(
+    const std::uint32_t transpose=0, const std::uint32_t ct_dim=0 /* unused */, const std::uint32_t rt_dim=0 /* unused */, const std::uint32_t kt_dim=0 /* unused */) {
+    _llk_unpack_AB_matmul_mop_config_(transpose>0);
 }
 
+inline void _llk_unpack_AB_matmul_(const std::uint32_t address_a, const std::uint32_t address_b) {
 
-inline void llk_unpack_AB_matmul_init(const std::uint32_t transpose=0) { llk_unpack_AB_matmul_mop_config(transpose>0); }
-
-inline void llk_unpack_AB_matmul(
-    std::uint32_t operandA, std::uint32_t operandB, std::uint32_t tile_index_a, std::uint32_t tile_index_b) {
-    std::uint32_t inputA = get_operand_id(operandA);
-    std::uint32_t inputB = get_operand_id(operandB);
-
-    std::uint32_t base_address_a = cb_interface[inputA].fifo_rd_ptr;
-    std::uint32_t offset_address_a = MUL_TILE_SIZE_AND_INDEX((uint)unpack_src_format[inputA], tile_index_a);
-    std::uint32_t base_address_b = cb_interface[inputB].fifo_rd_ptr;
-    std::uint32_t offset_address_b = MUL_TILE_SIZE_AND_INDEX((uint)unpack_src_format[inputB], tile_index_b);
-
-    // note: unpacker is programmed to automatically skip the tile header (+1)
-    // since there is no tile header, we need to -1 the address (in terms of 16B words), to offet unpacker's automatic +1
-    std::uint32_t address_a = base_address_a + offset_address_a - 1;
-    std::uint32_t address_b = base_address_b + offset_address_b - 1;
+    // Todo: do something with tile dim flags
+    volatile uint tt_reg_ptr *cfg = get_cfg_pointer();  // get pointer to registers for current state ID
```

Comment:
I believe hoisting this outside the function and before the matmul rt_dim/ct_dim loops did improve performance. 
@mywoodstock measured BERT-L on this branch at 411 sent/sec, so perhaps it's getting auto-hoisted in the refactored code.

Reply:
I think when we commonlize the code, we can try taking the cfg pointer specifically out and see if it helps performance for metal/buda, but I think the bigger impact was the address calculations being in the higher level loops:
```
    const std::uint32_t operandA_id = get_operand_id(operandA);
    const std::uint32_t operandB_id = get_operand_id(operandB);

    std::uint32_t base_address_a = cb_interface[operandA_id].fifo_rd_ptr - 1;
    std::uint32_t base_address_b = cb_interface[operandB_id].fifo_rd_ptr - 1;
    std::uint32_t unpA_src_format = (std::uint32_t)unpack_src_format[operandA_id];
    std::uint32_t unpB_src_format = (std::uint32_t)unpack_src_format[operandB_id];

    for (std::uint32_t rt=0; rt<rt_dim; rt++) {
        std::uint32_t offset_address_a = MUL_TILE_SIZE_AND_INDEX<true>(unpA_src_format, (tile_index_a + rt*kt_dim));
        std::uint32_t address_a = base_address_a + offset_address_a;

        for (std::uint32_t ct=0; ct<ct_dim; ct++) {

            std::uint32_t offset_address_b = MUL_TILE_SIZE_AND_INDEX<true>(unpB_src_format, (tile_index_b+ct));
            std::uint32_t address_b = base_address_b + offset_address_b;

            _llk_unpack_AB_matmul_(
                address_a,
                address_b
            );
```

because we did see before that being able to push those address calculations to replay buffers/outerloops in wormhole gave a lot of performance gains. 

But currently because llk_unpack_AB_matmul.h file is common for both buda/metal, and buda is still calculating by tile, then the cfg pointer will have to stay in the per tile calculation until we commonlize/back-propagate back to buda